worlds/factorio/data/mod_template/control.lua

```diff
@@ -660,11 +660,18 @@ commands.add_command("ap-get-technology", "Grant a technology, used by the Archi
     end
     local tech
     local force = game.forces["player"]
+    if call.parameter == nil then
+        game.print("ap-get-technology is only to be used by the Archipelago Factorio Client")
+        return
+    end
     chunks = split(call.parameter, "\t")
     local item_name = chunks[1]
     local index = chunks[2]
     local source = chunks[3] or "Archipelago"
-    if index == -1 then -- for coop sync and restoring from an older savegame
+    if index == nil then
+		game.print("ap-get-technology is only to be used by the Archipelago Factorio Client")
```

Comment:
indentation seems  to have gone wild  here.

Reply:
meant for the formatting to be spaces, not tabs.  Commited b08791b to resolve this.